website/static/js/newAndNoteworthyPlugin.js

```diff
@@ -0,0 +1,195 @@
+        popularPromise.then(function(result){
+            var numPopular = Math.min(result.data.length - 1, 4);
+            for (var l=0; l <= numPopular; l++) {
+                self.popularNodes().push(result.data[l]);
+                self.fetchContributors(result.data[l]);
+                  }
+            return popularPromise;
+        });
+
+        // Additional API call to fetch node link contributors
+        self.fetchContributors = function(nodeLink) {
+            var url = nodeLink.embeds.target_node.data.relationships.contributors.links.related.href;
+            var promise = m.request({method: 'GET', url : url, config: xhrconfig});
+            promise.then(function(result){
+                var contribNames = [];
+                result.data.forEach(function (contrib){
+                    contribNames.push(contrib.embeds.users.data.attributes.family_name);
+                });
+                var numContrib = result.links.meta.total;
+                var nodeId = nodeLink.id;
+                self.contributorsMapping[nodeId] = [contribNames, numContrib];
+            });
+        };
+
+        // Gets contrib family name for display
+        self.getFamilyName = function(i, node) {
+            return self.contributorsMapping[node.id][0][i];
+        };
+
+        // Returns name if one contrib, or adds et al if > 1
+        self.contribNameFormat = function(node, number) {
+            if (number === 1) {
+                return self.getFamilyName(0, node);
+            }
+            else if (number === 2) {
+                return self.getFamilyName(0, node) + ' and ' +
+                    self.getFamilyName(1, node);
+            }
+            else if (number === 3) {
+                return self.getFamilyName(0, node) + ', ' +
+                    self.getFamilyName(1, node) + ', and ' +
+                    self.getFamilyName(2, node);
+            }
+            else {
+                return self.getFamilyName(0, node) + ', ' +
+                    self.getFamilyName(1, node) + ', ' +
+                    self.getFamilyName(2, node) + ' + ' + (number - 3);
+            }
+        };
+
+        // Formats contrib names for display
+        self.getContributors = function (node) {
+            return self.contribNameFormat(node, self.contributorsMapping[node.id][1]);
+        };
+
+        // Grabs title for display
+        self.getTitle = function (node){
+            return node.embeds.target_node.data.attributes.title;
+        };
+
+        // Grabs description for display
+        self.getDescription = function(node){
+            var description = node.embeds.target_node.data.attributes.description;
+            if (description){
+                return description;
+            }
+            return m('p', {class: 'blank-line'});
+        };
+
+         // Onclick, directs user to project page
+        self.nodeDirect = function(node) {
+            location.href = '/' + node.embeds.target_node.data.id;
```

Comment:
These single line functions don't need to be functions, the onclick functions in the code are good places for these